Major changes to support latest Assetto Corsa version

Pringlez opened the pull request:
Changed the way track config loading is done, it relies on local images & json files instead of contentPath. This will be helpful for linux installation.

Fixed some HTML issues, tags, clean up & updated help page image.

Changed server.js inconsistencies related to quotes.

Updated messages displayed to user, nicer output :)

Tested on Windows, seems to all work fine. Will test on my Linux install later...

Pringlez committed:
Changed track UI loading & other changes

Changed the way track config loading is done, it relies on local images & json files instead of contentPath. This will be helpful for linux installation.

Fixed some HTML issues, tags, clean up & updated help page image.

Changed server.js inconsistencies related to quotes.

Updated messages displayed to user, nicer output :)

Tested on Windows, seems to all work fine. Will test on my Linux install later...

Pringlez commented:
Tested on Linux installation @jo3stevens , seems to work just fine. Track images load fine & config layouts of tracks.

Cele91 commented:
Negative. If I select "club" of Vallelunga, after refresh it appear "classic"

Pringlez commented:
I have a fix for that, I'm working on fixing other save issues when I get time

Pringlez committed:
Fixed a number of issues

More commits to come, to finish changes later

Drop track select fixed
Updated angular verion etc

jo3stevens commented:
Hey @Pringlez 

Also looks like you've put a whole load of the track JSON files into this project. The idea is that it loads them from the server content folder which would be updated by the user as and when AC do updates so we don't want them in this repo. It was also set up for a future addition of multiple tracks.

Some of the other changes don't really make sense to me either such as simple changes removing the min value from number text fields?

Pringlez commented:
Hey @jo3stevens 

I've removed the sun angle calculation because it doesn't seem to populate the UI HRs & MMs fields if for instance the sun angle is set to '-8' or lower, I'm still looking into.

The reason why I removed the 'min' value from that field is because it was preventing the form validating & saving the data from what I found.

The track & car content is not available on a remote Linux server, so this allows server admins use a 'local' copy of the content on a remote machine that has no access to an installation of Assetto Corsa. You can disable this by setting the 'useLocalContent' to false.

I've updated the Angular version & removed some JS files not in use anymore.

I plan to clean up these commits with another one, I've just noticed some logs got in there....

jo3stevens commented:
Hey @Pringlez 

I'd rather not merge this request in. I'll just fix the binding issues when I get some time though there were no binding issues before the new design so not sure what caused them.

I also don't want all of the track content in this project. Why does linux not have it? Surley it just needs a script similar to the .bat file that scrapes the required data and dumps it in the server folder?

Cheers
Joe

Pringlez committed:
Update for 1.14 AC patch

Added the new configuration settings for wind & time of day

Fixed a issues with 'Locked Entry List', 'Loop Session', 'Pickup Mode Enabled' & 'Register to Lobby' not switching to the correct button value

Reverted Sun Angle calculation & added in option to either set the sun angle directly or with hours & mins fields

Updated readme & version number

Working on local Windows machine

Pringlez commented:
I believe I fixed most of the saving issues with my last commit just there.

The local content I added is only about 5mb, only small ini & json files were added. It's a real headache to get this content onto a remote Linux server by copying stuff over ssh. I know it may look messy & requires some ongoing maintenance as new tracks and cars get added, but it would be a hell of alot easier for Linux hosting 👍 

John W

Pringlez committed:
Fix for content path

Pringlez committed:
Small fix for 'contentPath' for cars

Small fix for cars API because local content for cars has not been commited to repo.

hmehta committed:
Implement content management for tracks and cars

* Add support for uploading both single and multilayout tracks using
  drag & drop
* Add support for removing existing tracks and cars

hmehta committed:
Add uuid dependency

hmehta committed:
Add API for templates

hmehta committed:
Add default server configuration template

hmehta committed:
Add TemplateService

hmehta committed:
Add TemplatesCtrl controller

hmehta committed:
Add Templates HTML

Pringlez committed:
Merge pull request #8 from hmehta/add-template-support

Add template support

hmehta committed:
Add Laguna Seca track included in AC v1.16

hmehta committed:
Update for v1.16

Pringlez committed:
Merge pull request #9 from hmehta/v1.16

Update content for v1.16, add linux frontend content generation scripts for cars / skins

Pringlez committed:
Merge pull request #10 from hmehta/hmehta-manage-tracks-and-cars

Tracks and cars management, merging to master

Tested on Linux, seems to work without issues, templates work fine + all other functionality seems to work fine. Drag & drop feature works quite well!

Some code indentation clean up required in some files

Pringlez committed:
Updating README & other small updates
To keep things consistent & bumped version number

Pringlez commented:
Hey @jo3stevens,

It's been awhile, would you like to take a look at the latest version of the fork? There has been quite a few updates since last time :+1: 

Cheers
John W

hmehta committed:
Make entry list editable in-place

This helps administrators to more easily configure Steam GUIDs, ballasts
and such for drivers participating regularly to a series of races.

hmehta committed:
Add Restrictor support

hmehta committed:
Display last 2000 lines of acServer logs in Web UI

hmehta committed:
Add support for sTracker on Linux

Pringlez committed:
Merge pull request #13 from hmehta/hmehta-linux-stracker-support

Add support for sTracker on Linux

Pringlez committed:
Merge pull request #12 from hmehta/hmehta-show-server-logs

Show server logs

Pringlez committed:
Merge pull request #11 from hmehta/hmehta-enhance-entry-list

Enhance entry list

Pringlez committed:
Intial commit for docker build file
Added new docker build to allow users host application in containers
Updated readme

Pringlez committed:
Updated docker file to fix pm2 start issues

Pringlez committed:
Removing NVM from docker image, app should start on boot now.
Updated readme

Pringlez committed:
Update readme's & docker file

Pringlez committed:
Changes to docker file to include meta data
For build badges

Pringlez committed:
Removing steamcmd install from the image
Simplifying the build image

Pringlez committed:
Updating readme's

Pringlez committed:
Update to docker build file & readme's

Pringlez committed:
Update to readme's

Pringlez committed:
Change to docker file & readme's

TylerBre commented:
@Pringlez issue was with a node fn ```fs.copyFileSync```

https://nodejs.org/api/fs.html#fs_fs_copyfilesync_src_dest_flags

essentially this makes your code have a hard requirement for node >= v8.5.0

I would recommend putting that in package.json via engines block and engineStrict:

http://www.marcusoft.net/2015/03/packagejson-and-engines-and-enginestrict.html

TylerBre commented:
@jo3stevens will this ever be merged in? It has a number of good bug fixes and useful functionality.

I want to put this into a docker image but I don't like that it's just a pr, wish it was on a stable branch, etc.

Pringlez commented:
@TylerBre I'm not sure if the engines block and engineStrict is supported anymore, since node version 3+ I think it has been deprecated. Is that correct?
Maybe I should update the readme to highlight the issue.

TylerBre commented:
@Pringlez engines definitely is, https://docs.npmjs.com/files/package.json#engines

Personally I've never used enginesStrict, but I've used npm scripts that will check your locally installed node version and exit the process if your node version doesn't match up with what's in package.json

Or you could just shim ```copyFileSync``` with a node module or something

Pringlez commented:
@TylerBre Feel free to implement a fix & pull request :+1: . I could look into later if I get time. I'll just update the readme for now. Good work finding the issue there. Maybe @hmehta could help you out since he implemented the feature :+1:

Pringlez committed:
Updating readme's for recent found issue

Node version 8.5.0 or newer is now preferred

Commit:
added fs-copy-file-sync

TylerBre commented:
@Pringlez done:

https://github.com/Pringlez/ACServerManager/pull/14

Pringlez committed:
Updating screenshots

Pringlez committed:
Merge pull request #14 from TylerBre/master

Using fs-copy-file-sync over fs.copyFileSync for backwards compatability
